nixos/modules/virtualisation/qemu-guest-agent.nix

```diff
@@ -0,0 +1,39 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.qemuGuest;
+in {
+
+  options.services.qemuGuest = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Whether to enable the dspam spam filter.";
+      };
+  };
+
+  config = mkIf cfg.enable (
+      mkMerge [
+    {
+
+      services.udev.extraRules = ''
+        SUBSYSTEM=="virtio-ports", ATTR{name}=="org.qemu.guest_agent.0", TAG+="systemd" ENV{SYSTEMD_WANTS}="qemu-guest-agent.service"
+      '';
+
+      environment.systemPackages = [ pkgs.qemu ];
+
+      systemd.services.qemu-guest-agent = {
+        description = "Run the QEMU Guest Agent";
+        path = [ pkgs.qemu ];
+        serviceConfig = {
+          ExecStart = "${pkgs.kvm}/bin/qemu-ga";
```

Comment:
I suggest to use `pkgs.qemu.ga` or `pkgs.kvm.ga` here as discussed earlier. Should be consistent with what you use in `environment.systemPackages` above.

Reply:
I suppose I could get rid of the environment.systemPackages too